### PR TITLE
Parametrized Product Upgrades

### PR DESCRIPTION
**Pull request by jyejare**
- Making upgrade fab tasks parametrized to take satellite hostname from CI for multi satellite upgrades
- Some minor improvements to code

### CODE REVIEW

File: upgrade/helpers/tools.py [hunk 1]
```diff
@@ -233,21 +237,31 @@ def create_setup_dict(setups_dict):
     :param dict setups_dict: Dictionary of all return value of
     setup_products_for_upgrade
     """
-    with open('product_setup', 'w') as pref:
-        json.dump(setups_dict, pref)
+    data = {}
+
+    if os.path.exists(setup_data_file):
```
Comment (mshriver) [inline upgrade/helpers/tools.py:242]: consider using `pathlib.Path` instances for these file interactions.

**Comment by devendra104** [comment]
@jyejare Please resolve the conflicts.

**Comment by jyejare** [comment]
@devendra104 Rebased !

File: upgrade/helpers/tools.py (hunk 1 above)
Comment (jyejare) [reply upgrade/helpers/tools.py]: Done !

**Comment by jyejare** [comment]
@devendra104 @mshriver Ready for 2nd round of review.

**Review by mshriver — APPROVED** [review]

**Review by devendra104 — APPROVED** [review]
ACK

**Pull request merged**